framework/src/kernels/Kernel.C

```diff
@@ -41,21 +41,62 @@ Kernel::Kernel(const InputParameters & parameters)
     _u_dot(_var.uDot()),
     _du_dot_du(_var.duDotDu())
 {
+  auto & vector_tag_names = getParam<MultiMooseEnum>("vector_tags");
```

Comment:
We will possibly need the same thing for BCs, DGKernels, etc.

Reply:
We want a tagging interface, and then BCs, DGKernel, etc. should derive from the tagging interface.